src/CoreVideo/CVMetalTextureCache.cs

```diff
@@ -88,6 +88,37 @@ public static CVMetalTextureCache FromDevice (IMTLDevice metalDevice)
 			return null;
 		}
 
+#if XAMCORE_2_0
+		public CVMetalTextureCache (IMTLDevice metalDevice, CVMetalTextureAttributes textureAttributes)
+		{
+			if (metalDevice == null)
+				throw new ArgumentNullException (nameof (metalDevice));
+
+			if (CVMetalTextureCacheCreate (IntPtr.Zero,
+								IntPtr.Zero, /* change one day to support cache attributes */
+								metalDevice.Handle,
+								textureAttributes?.Dictionary.Handle ?? IntPtr.Zero,
+								out handle) == 0)
+				return;
+
+			throw new Exception ("Could not create the texture cache");
```

Comment:
I think it would be an improvement to put the return value in the exception message (and if possible find a string value for the error code), that makes it easier to diagnose failures.